src/ip_voter.cpp

```diff
@@ -173,14 +173,31 @@ namespace libtorrent
 		return true;
 	}
 
-	bool external_ip::cast_vote(address const& ip, int source_type, address const& source)
+	external_ip::external_ip(address local4, address global4
+		, address local6, address global6)
 	{
-		return m_vote_group[ip.is_v6()].cast_vote(ip, source_type, source);
+		m_addresses[0][0] = global4;
+		m_addresses[0][1] = global6;
+		m_addresses[1][0] = local4;
+		m_addresses[1][1] = local6;
+
+#if TORRENT_USE_IPV6
+		if (m_addresses[0][1] == address_v4())
```

Comment:
I wonder if it would work out nicely to have a function: ``address ensure_v6(address a) { return a == address_v4() ? address_v6() : a; }``

and use that in the aggregate initialization